token-lending/program/src/processor.rs

```diff
@@ -1586,31 +1586,30 @@ fn process_flash_loan(program_id: &Pubkey, amount: u64, accounts: &[AccountInfo]
     }
     reserve.liquidity.borrow(amount)?;
     spl_token_transfer(TokenTransferParams {
-        source: reserve_liquidity_supply_info.clone(),
+        source: source_liquidity_info.clone(),
         destination: destination_liquidity_info.clone(),
         amount,
         authority: derived_lending_market_account_info.clone(),
         authority_signer_seeds,
         token_program: token_program_id.clone(),
     })?;
 
-    let (origination_fee, host_fee) = reserve.config.fees.calculate_flash_loan_fees(amount)?;
+    let (origination_fee, host_fee) = reserve.config.fees.calculate_flash_loan_fee(amount)?;
```

Comment:
This should be `calculate_flash_loan_fees` as it calculates not one fee but two.